src/MongoDB.Driver.Legacy/MongoServerSettings.cs

```diff
@@ -1114,5 +1127,15 @@ internal ClusterKey ToClusterKey()
                 _waitQueueSize,
                 _waitQueueTimeout);
         }
+
+        private void ThrowIfSettingsAreInvalid()
+        {
+            if (AllowInsecureTls && SslSettings != null && SslSettings.CheckCertificateRevocation)
```

Comment:
It's not a big deal but we usually use the backing fields directly instead of the properties from inside the class.

Reply:
Fixed!